compose/material/material/src/desktopMain/kotlin/androidx/compose/material/DesktopMenu.desktop.kt

```diff
@@ -256,6 +256,10 @@ private data class DesktopDropdownMenuPositionProvider(
             y = anchorBounds.bottom + contentOffsetY
         }
 
+        if (y + popupContentSize.height > windowSize.height) {
+            y = windowSize.height - popupContentSize.height
```

Comment:
Better to add this check too:
```
if (y < 0) {
    y = 0
}
```

We have a cut off popup in run3:

![image](https://user-images.githubusercontent.com/5963351/143845525-300cb8bf-477f-4f4a-91bf-9e1e24fba2b4.png)

Reply:
Done.